allauth/socialaccount/providers/microsoft/provider.py

```diff
@@ -8,7 +8,10 @@ class MicrosoftGraphAccount(ProviderAccount):
 
     def to_str(self):
         name = self.account.extra_data.get('displayName')
-        if name.strip() != '':
+        if not name is None:
+            if name.strip() != '':
+                return name
+        else:
             return name
         return super(MicrosoftGraphAccount, self).to_str()
```

Comment:
With these changes you never end up in the `return super()...` statement. So I think the only think that needs to be changed is the `if`. Change it into `if name and name.strip() != '':`

Reply:
Thanks man, you were right ;) Just made the change and it passed test coverage !